app/modules/mqtt.c

```diff
@@ -281,11 +310,109 @@ static void mqtt_socket_received(void *arg, char *pdata, unsigned short len)
   if(mud == NULL)
     return;
 
+  switch(mud->mqtt_state.recv_buffer_state) {
+    case MQTT_RECV_NORMAL:
+      // No previous buffer.
+      break;
+    case MQTT_RECV_BUFFERING_SHORT:
+      // Last buffer had so few byte that we could not determine message length.
+      // Store in a local heap buffer and operate on this, as if was the regular pdata buffer.
+      // Avoids having to repeat message size/overflow logic.
+      temp_pdata = c_zalloc(mud->mqtt_state.recv_buffer_size + len);
+      if(temp_pdata == NULL) {
+        NODE_DBG("MQTT[buffering-short]: Failed to allocate %u bytes, disconnecting...\n", mud->mqtt_state.recv_buffer_size + len);
```

Comment:
Sorry, I'm not very familiar with this module; will this disconnection be exposed to the Lua callbacks?  (Does the espconn layer call us back on disconnection, for example?)

Reply:
Yes, as soon as the MQTT socket is connected, it registers an disconcb (in `mqtt_socket_connected`) which will trigger the `offline` callback (as bound via `on("offline", ..)`.

Reply:
Rad.